### ISSUE TITLE
BUG: `constants.pc` definition does not use small angle approximation

### ISSUE THREAD

**Issue by dhomeier** [astropy/astropy#10568]
### Description
As reported on the [mailing list](https://mail.python.org/pipermail/astropy/2020-July/004772.html) the definition of the parsec in https://github.com/astropy/astropy/blob/28349153fcc25a9ca01bb5e563c86177c50aed1d/astropy/constants/iau2015.py#L29
uses the exact distance for a 1"-parallax object `c.au / np.tan(u.arcsec.to('radian'))` rather than the small-angle approximation `c.au / u.arcsec.to('radian')`. However the [IAU 2015 resolution B2](https://www.iau.org/static/resolutions/IAU2015_English.pdf) states (footnote 4) that 
> The parsec is defined as exactly (648 000/π) au (e.g. Cox 2000, Binney & Tremaine 2008).

([small-angle approximation](https://en.wikipedia.org/wiki/Parsec#cite_note-11)).
### Expected behavior
```
from astropy import constants as apc
from astropy import units as apu
print(apc.au.value * 180 * 3600 / np.pi)
3.085677581491367e+16 
```
### Actual behavior
```
print(apc.pc.value)
print(apc.au.value / np.tan(np.pi / (180 * 3600)))
3.0856775814671916e+16
3.0856775814671916e+16
```

Fixing the value in `iau2015.py` is trivial; perhaps to be discussed if the change should also be copied into `iau2012.py` (which implicitly certainly assumed the same definition, but did not state this).
Also `equivalencies=u.parallax()` will probably need to be adapted as well.

**Comment by pllim** [comment]
cc @stargaser

**Comment by stargaser** [comment]
Thank you @dhomeier for putting in this issue and replying to the email. You beat me to it!

**Comment by dhomeier** [comment]
Do you want to start a PR @stargaser ? As indicated I have not really looked into what to do about the equivalency; it might be straightforward as well...

**Comment by pllim** [comment]
Hopefully the tests will show us that the results will not change in a way that `FLOAT_CMP` or `allclose` cares, otherwise we might be in trouble. 😅 🤞

**Comment by stargaser** [comment]
I probably won't get to a PR for a few hours or maybe a couple of days...if someone else wants to start one before I get to it.

**Comment by dhomeier** [comment]
> Hopefully the tests will show us that the results will not change in a way that `FLOAT_CMP` or `allclose` cares

I you might be a bit too optimistic. As Jenny Hatchell noted, the discrepancy is O(10^5) float64 precisions; could still pass the `allclose` defaults, but what does `FLOAT_CMP` expect?

**Comment by pllim** [comment]
`FLOAT_CMP` defaults at https://github.com/astropy/pytest-doctestplus/blob/4028a7673ac323842dce09dbb054996c90cc91de/pytest_doctestplus/plugin.py#L105-L111 . There are options to customize but I don't think we want to do that just for this case.

**Comment by dhomeier** [comment]
Same as the `np.allclose` defaults; should be safe then. 😅

**Comment by dhomeier** [comment]
If I read `parallax_converter` correctly, it is already using 1 / angle, so it should be down to the `pc` and `kpc` definitions indeed. I am off for a few hours nonetheless, but will look into it later today.

**Comment by stargaser** [comment]
Thanks @dhomeier for getting into the details already. I have a proposal deadline in just over 48 hours. @mhvk and I can take a review role on this one.

**Comment by pllim** [comment]
Fixed in #10569. Thanks, everyone!
